plugins/scancode-fingerprint/src/plugin_fingerprint/fingerprint.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+import binascii
+from bitarray import bitarray
+from bitarray import bitdiff
+from licensedcode.tokenize import ngrams
+import hashlib
+
+HASH_LENGTH = 128
+SHINGLE_LENGTH = 3
+
+class Simhash:
+    """
+    Fingerprint class to generate fingerprints for files used for similarity matching
+    """
+    def __init__(self):
+        self.tokens = []
+
```

Comment:
Minor formatting issue: there should be one new line between methods and two new lines between functions and classes.